MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/mas/TurfInsideActivity.java

```diff
@@ -0,0 +1,238 @@
+import android.os.Bundle;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.mapbox.mapboxandroiddemo.R;
+import com.mapbox.mapboxsdk.Mapbox;
+import com.mapbox.mapboxsdk.annotations.Marker;
+import com.mapbox.mapboxsdk.annotations.MarkerViewOptions;
+import com.mapbox.mapboxsdk.annotations.Polygon;
+import com.mapbox.mapboxsdk.annotations.PolygonOptions;
+import com.mapbox.mapboxsdk.geometry.LatLng;
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
+import com.mapbox.services.api.utils.turf.TurfJoins;
+import com.mapbox.services.commons.models.Position;
+import com.mapbox.services.commons.utils.TextUtils;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TurfInsideActivity extends AppCompatActivity {
+
+  private MapView mapView;
+  private MapboxMap map;
+  private Polygon polygon;
+  private Marker withinMarker;
+
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the account manager
+    setContentView(R.layout.activity_turf_inside);
+
+
+    mapView = (MapView) findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(new OnMapReadyCallback() {
+      @Override
+      public void onMapReady(MapboxMap mapboxMap) {
+
+        map = mapboxMap;
+
+        // Draw and display GeoJSON polygon on map
+        new DrawGeoJson().execute();
+
+        onMapClick();
+
+      }
+    });
+
+    Toast.makeText(this, getString(R.string.tap_on_map_turf_inside_instruction), Toast.LENGTH_SHORT).show();
+
+  }
+
+  // Add the mapView lifecycle to the activity's lifecycle methods
+  @Override
+  protected void onStart() {
+    super.onStart();
+    mapView.onStart();
+  }
+
+  @Override
+  public void onResume() {
+    super.onResume();
+    mapView.onResume();
+  }
+
+  @Override
+  public void onPause() {
+    super.onPause();
+    mapView.onPause();
+  }
+
+  @Override
+  protected void onStop() {
+    super.onStop();
+    mapView.onStop();
+  }
+
+  @Override
+  public void onLowMemory() {
+    super.onLowMemory();
+    mapView.onLowMemory();
+  }
+
+  @Override
+  protected void onDestroy() {
+    super.onDestroy();
+    mapView.onDestroy();
+  }
+
+  @Override
+  protected void onSaveInstanceState(Bundle outState) {
+    super.onSaveInstanceState(outState);
+    mapView.onSaveInstanceState(outState);
+  }
+
+  public void onMapClick() {
```

Comment:
Looks like this method only handles the onclick? in this case you can just add it as an implementation like so:

```java
... extends AppCompatActivity implements MapboxMap.OnMapClickListener {
```

```
 @Override
  public void onMapReady(MapboxMap mapboxMap) {
    mapboxMap.setOnMapClickListener(this);
```

```
@Override
  public void onMapClick(@NonNull LatLng point) {
}
```